### PR TITLE
Update version guards to deal for new query profile collector fields

### PR DESCRIPTION
**Pull request by ticheng-aws**
### Description
https://github.com/opensearch-project/OpenSearch/pull/7975

### Related Issues
Resolves https://github.com/opensearch-project/OpenSearch/issues/7355
https://github.com/opensearch-project/OpenSearch/pull/7975/files/fa83dc5a5e65953ce8a87e4adbee0f4ca2f2e230#r1223642855

### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [X] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -104,6 +104,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Compress and cache cluster state during validate join request ([#7321](https://github.com/opensearch-project/OpenSearch/pull/7321))
 - [Snapshot Interop] Add Changes in Create Snapshot Flow for remote store interoperability. ([#7118](https://github.com/opensearch-project/OpenSearch/pull/7118))
 - Add new query profile collector fields with concurrent search execution ([#7898](https://github.com/opensearch-project/OpenSearch/pull/7898))
+- Update version guards to deal for new query profile collector fields ([#7979](https://github.com/opensearch-project/OpenSearch/pull/7979))
```
Comment (reta) [inline CHANGELOG.md:107]: We don't need the changelog entry (this is essentially the part of the `Add new query profile collector fields with concurrent search execution`), thank you

File: server/src/main/java/org/opensearch/search/profile/query/CollectorResult.java [hunk 2]
```diff
@@ -164,7 +164,7 @@ public CollectorResult(StreamInput in) throws IOException {
             CollectorResult child = new CollectorResult(in);
             this.children.add(child);
         }
-        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+        if (in.getVersion().onOrAfter(Version.V_2_9_0)) {
```
Comment (reta) [inline server/src/main/java/org/opensearch/search/profile/query/CollectorResult.java:167]: 👍

**Comment by reta** [comment]
The failure is fine, one we merge the backport to `2.x`, the build will be green

File: CHANGELOG.md (hunk 1 above)
Comment (ticheng-aws) [reply CHANGELOG.md]: Got it. It has been addressed.

**Comment by ticheng-aws** [comment]
> The failure is fine, once we merge the backport to `2.x`, the build will be green

Thanks, @reta. Please kindly review it again.

**Review by reta — APPROVED** [review]

**Pull request merged**